### PR TITLE
[hold] Runtimeclass API

### PR DESCRIPTION
**Pull request by tallclair**
Adds (but doesn't implement) the RuntimeClass API proposed in https://github.com/kubernetes/community/blob/master/keps/sig-node/0014-runtime-class.md

For https://github.com/kubernetes/features/issues/585

**Special notes for your reviewer**:

**Release note**:
```release-note
RuntimeClass is a new API resource for defining different classes of runtimes that may be used to run containers in the cluster. Pods can select a RunitmeClass to use via the RuntimeClassName field. This feature is in alpha, and the RuntimeClass feature gate must be enabled in order to use it.
```

/sig node
/kind api-change
/priority important-soon
/milestone v1.12

### CODE REVIEW

**Comment by tallclair** [comment]
/hold
This feature still needs sig-architecture review.

/cc @dchen1107 @derekwaynecarr

File: staging/src/k8s.io/api/node/v1alpha1/types.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+// RuntimeClassList is a list of RuntimeClass objects.
+type RuntimeClassList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Items is a list of schema objects.
```
Comment (CaoShuFeng) [inline staging/src/k8s.io/api/node/v1alpha1/types.go]: Items is a list of `RuntimeClass`?

**Comment by tallclair** [comment]
Closing to separate pod changes & new API.

**Pull request closed without merging**